### PR TITLE
Compiling ipynb now looks for default jupyter configuration

### PR DESCRIPTION
**Pull request by ralsina**
This is a followup from #2650

For those, like @knowsuchagency, who use the jupyter extensions, specifically python markdown, this makes it so Nikola will automatically look to see if users have edited the configuration files nbconvert normally looks at when compiling notebooks itself.

Thus if you have a markdown cell that jupyter normally converts to 2 from {{ 1 + 1 }} in your notebook, you'll see the former rather than the later.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ New in master
 Features
 --------
 
+* Add Jupyter config as dependency for jupyter posts (by @knowsuchagency)
```
Comment (Kwpolska) [inline CHANGES.txt]: What’s the purpose of this attribution?

File: nikola/plugins/compile/ipynb.py [hunk 2]
```diff
@@ -190,3 +192,30 @@ def create_post(self, path, **kw):
...
+        return config
+
+    for parent in jupyter_config_path():
+        try:
+            for file in os.listdir(parent):
+                if 'nbconvert' in file and file.endswith('.json'):
+                    abs_path = os.path.join(parent, file)
+                    with open(abs_path) as config_file:
+                        config.update(json.load(config_file))
+        except FileNotFoundError:
```
Comment (Kwpolska) [inline nikola/plugins/compile/ipynb.py]: `except OSError:` for py2 compatibility

File: CHANGES.txt (hunk 1 above)
Comment (ralsina) [inline CHANGES.txt]: New contributor encouragement :-)

File: nikola/plugins/compile/ipynb.py (hunk 2 above)
Comment (ralsina) [inline nikola/plugins/compile/ipynb.py]: Done

**Pull request merged**